### PR TITLE
fixes #3101 and #3120 - missing features in api around images

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

**Comment by GregSutcliffe** [comment]
When dependency packages change, can we get either a PR to foreman-infra/deb/develop or a least a notification so I can do it?

File: Gemfile [hunk 1]
```diff
@@ -14,11 +14,15 @@ gem "ancestry", "~> 1.3"
 gem 'scoped_search', '>= 2.5'
 gem 'net-ldap'
 gem 'uuidtools'
-gem "apipie-rails", "~> 0.0.22"
+gem "apipie-rails", "~> 0.0.23"
 gem 'rabl', '>= 0.7.5'
 gem 'oauth'
 gem 'foreigner', '~> 1.4.2'
 
+gem 'thin'
```
Comment (domcleal) [inline Gemfile]: Could you keep these in bundler.d/Gemfile.local.rb please? :)

**Comment by domcleal** [comment]
Tests ok, thanks.  Why the apipie bump, is there a feature/bug fix we need?  If so, could you file a ticket and we'll get that included.

**Comment by domcleal** [comment]
Oh, the Jenkins tests failed - you need to add this route to an appropriate permission in app/services/foreman/access_permissions.rb.

File: app/controllers/api/v2/compute_resources_controller.rb [hunk 2]
```diff
@@ -15,6 +17,12 @@ def show
         render :template => "api/v1/compute_resources/show"
       end
 
+      api :GET, "/compute_resources/:compute_resource_id/available_images/", "List available images for a compute resource."
+      param :id, :identifier, :required => true
```
Comment (domcleal) [inline app/controllers/api/v2/compute_resources_controller.rb]: mismatch between IDs here, one's `:compute_resource_id` above and this is :id?

File: app/views/api/v1/images/show.json.rabl [hunk 3]
```diff
@@ -1,3 +1,4 @@
 object @image
 
 attributes :id, :operatingsystem_id, :compute_resource_id, :architecture_id, :uuid, :username, :name, :created_at, :updated_at
+attribute :iam_role
```
Comment (ohadlevy) [inline app/views/api/v1/images/show.json.rabl]: only if CR==ec2?

**Comment by tstrachota** [comment]
Permission added.
The latest version of apipie-rails includes some importatnt fixes. Details in the ticket 3120.

**Comment by domcleal** [comment]
Thanks @tstrachota, merged as d484958 (for 1.4) and 36c04c36 (for 1.3).

**Pull request closed without merging**